### PR TITLE
Add UniteTest Data Generation Class, Add HashField UnitTest and Pass it.

### PR DESCRIPTION
**Pull request by Zrealshadow**
- Add `TestTable` class, which can load table data from csv for unitTest.
- Add `TestTableTest`, which is a unitTest for TestTable class. Print the loaded table and compare with truth data manually.
- Add `HashFieldTest` class, which involves 4 class, `HashMetaFieldRS`, `HashMetaFieldWS`, `HashFieldWS`, `CoolFieldRS`. In Unittest, generate data and write it into buffer with writestore part; then read data from this buffer with readstore part. Compare the expected value and actual value

### CODE REVIEW

**Comment by Zrealshadow** [comment]
Fix the untracked file.
Clean commit log and rebase it.
plz, review and merge  
@hugy718

File: cool-core/src/test/java/com/nus/cool/core/io/store/HashFieldTest.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    /**
+     * HashFiledTest : Conversion between WriteStore and ReadStore
+     * 
+     * @param fieldName
+     * @param fType
+     * @throws IOException
+     */
+    @Test(dataProvider = "HashFieldTestDP")
+    public void HashFiledTest(String fieldName, FieldType fType) throws IOException {
```
[the remark this answers is not in the document]
Comment (Zrealshadow) [reply cool-core/src/test/java/com/nus/cool/core/io/store/HashFieldTest.java]: fix: Use HashFieldUnitTest

**Comment by hugy718** [comment]
works at my side. Maybe we shall place the TestData into the corresponding src/test/resources

**Comment by Zrealshadow** [comment]
fix it

**Pull request merged**